impedance/tests/test_circuit_elements.py

```diff
@@ -1,37 +1,47 @@
-from impedance.models.circuits.elements import R, C, L, W, A, E, G, T, s, p
+import string
 import numpy as np
 
+from impedance.models.circuits.elements import circuit_elements, s, p
+
 
 def test_all():
-    funcs = [R, C, L, W, A, E, G, T]
     freqs = [0.001, 1.0, 1000]
-    correct_vals = [[0.1, 0.1, 0.1],
-                    [-1591.5494309189532j,
-                     -1.5915494309189535j,
-                     -0.0015915494309189533j],
-                    [0.000628319j, 0.628319j, 628.319j],
-                    [(0.033333332999112786-79.57747433847442j),
-                     (0.03300437046673635-0.08232866785870396j),
-                     (0.0019947114020071634-0.0019947114020071634j)],
-                    [(1.26156626-1.26156626j),
-                     (0.03989423-0.03989423j),
-                     (0.00126157-0.00126157j)],
-                    [(26.216236841407248-8.51817171087997j),
-                     (6.585220960717244-2.139667994182814j),
-                     (1.6541327179718126-0.537460300252313j)],
-                    [(0.09999994078244179-0.00006283179105931961j),
-                     (0.07107755021941357-0.03427465211788068j),
-                     (0.00199550459845528-0.0019939172581851707j)],
-                    [(1.00041-0.00837309j),
-                    (0.0156037-0.114062j),
-                    (0.00141056-0.00141039j)]]
+    correct_vals = {'R': [0.1, 0.1, 0.1],
+                    'C': [-1591.5494309189532j,
+                          -1.5915494309189535j,
+                          -0.0015915494309189533j],
+                    'L': [0.000628319j, 0.628319j, 628.319j],
+                    'W': [(0.033333332999112786-79.57747433847442j),
+                          (0.03300437046673635-0.08232866785870396j),
+                          (0.0019947114020071634-0.0019947114020071634j)],
+                    'A': [(1.26156626-1.26156626j),
+                          (0.03989423-0.03989423j),
+                          (0.00126157-0.00126157j)],
+                    'E': [(26.216236841407248-8.51817171087997j),
+                          (6.585220960717244-2.139667994182814j),
+                          (1.6541327179718126-0.537460300252313j)],
+                    'G': [(0.09999994078244179-0.00006283179105931961j),
+                          (0.07107755021941357-0.03427465211788068j),
+                          (0.00199550459845528-0.0019939172581851707j)],
+                    'T': [(1.00041-0.00837309j),
+                          (0.0156037-0.114062j),
```

Comment:
added a test for `K`, ignoring `s`, and `p`, and using the num_params attribute of the function wrapper instead of an inputs dictionary
```
                   'K': [(1e-1-2e-5j),
                          (9.6154e-02-1.9231e-02j),
                          (2.5e-6-5e-4j)],
                    'RR': [10, 10, 10]}
    input_vals = [0.1, 0.2, 0.3, 0.4]
    for key, f in circuit_elements.items():
        if f.__name__ not in ['s', 'p']:
            num_inputs = f.num_params
            val = f(input_vals[:num_inputs], freqs)
```